diff --color -Naur setuptools-58.1.0.orig/setuptools/_distutils/command/install.py setuptools-58.1.0/setuptools/_distutils/command/install.py
--- setuptools-58.1.0.orig/setuptools/_distutils/command/install.py	2021-09-21 23:00:41.000000000 -0300
+++ setuptools-58.1.0/setuptools/_distutils/command/install.py	2021-09-23 14:54:46.089032596 -0300
@@ -44,19 +44,13 @@
         },
     'nt': WINDOWS_SCHEME,
     'pypy': {
-        'purelib': '$base/site-packages',
-        'platlib': '$base/site-packages',
-        'headers': '$base/include/$dist_name',
+        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'platlib': '$platbase/lib/python$py_version_short/site-packages',
+        'headers': '$base/include/python$py_version_short$abiflags/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
-    'pypy_nt': {
-        'purelib': '$base/site-packages',
-        'platlib': '$base/site-packages',
-        'headers': '$base/include/$dist_name',
-        'scripts': '$base/Scripts',
-        'data'   : '$base',
-        },
+    'pypy_nt': WINDOWS_SCHEME,
     }
 
 # user site schemes
diff --color -Naur setuptools-58.1.0.orig/setuptools/_distutils/sysconfig.py setuptools-58.1.0/setuptools/_distutils/sysconfig.py
--- setuptools-58.1.0.orig/setuptools/_distutils/sysconfig.py	2021-09-21 23:00:41.000000000 -0300
+++ setuptools-58.1.0/setuptools/_distutils/sysconfig.py	2021-09-23 14:56:17.238623626 -0300
@@ -144,19 +144,16 @@
     sys.base_exec_prefix -- i.e., ignore 'plat_specific'.
     """
 
-    if IS_PYPY and sys.version_info < (3, 8):
-        # PyPy-specific schema
-        if prefix is None:
-            prefix = PREFIX
-        if standard_lib:
-            return os.path.join(prefix, "lib-python", sys.version[0])
-        return os.path.join(prefix, 'site-packages')
-
     if prefix is None:
         if standard_lib:
             prefix = plat_specific and BASE_EXEC_PREFIX or BASE_PREFIX
         else:
             prefix = plat_specific and EXEC_PREFIX or PREFIX
+    
+    if IS_PYPY:
+        # PyPy-specific schema
+        if standard_lib:
+            return os.path.join(prefix, "lib-python", sys.version[0])
 
     if os.name == "posix":
         if plat_specific or standard_lib:
